tsl/src/continuous_aggs/invalidation.c

```diff
@@ -1217,7 +1229,19 @@ remote_invalidation_process_hypertable_log(int32 mat_hypertable_id, int32 raw_hy
 		elog(ERROR, "function was not provided with a valid distributed hypertable id");
 
 	DistCmdResult *result;
-	List *data_node_list = ts_hypertable_get_data_node_name_list(ht);
+	List *data_node_list;
+	/*
+	 * Run the cmd on available data nodes only. In case of DML happening to non-available
+	 * DNs the metadata on the AN will be removed for stale chunks on that unavailable
+	 * DN
+	 *
+	 * Do this only if the hypertable is replicated. Otherwise use all DNs
+	 */
+	if (ht->fd.replication_factor > 1)
+		data_node_list = ts_hypertable_get_available_data_node_names(ht, true);
+	else
+		data_node_list = ts_hypertable_get_data_node_name_list(ht);
+
```

Comment:
I don't think this is the right thing to do.

When refreshing we need to get the ranges to refresh. If a DN is down we might not get those invalidated ranges from that node, which means we might not refresh in that range.

This would only work if we can guarantee that the same invalidations exists on another data node, which would typically be the case if all the chunks in the range has a replica on another data node. 

But this just gets the available data nodes and doesn't check if all chunks in the range has a replica on the available data nodes.  For now, maybe it is just better to fail here.

I guess another option is to assume the whole range is invalid if a node is down and then just leave the invalidations on the unavailable nodes.

Reply:
changes reverted for now.